<?xml version="1.0"?>
<launch>
    <arg name="number" default=""/>
    <arg name="sim" default="true"/>
    <group ns="bebop10$(arg number)">
        <rosparam file="$(find bebop_aruco)/config/agent10$(arg number).yaml" command="load" ns="" />
        <include file="$(find bebop_aruco)/launch/agent_base.launch">
            <arg name="number" value="$(arg number)"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        
        
        <node pkg="bebop_hatanaka_base" type="posestamped2tf.py" name="vrpn_posestamped2tf" output="log" respawn="false" respawn_delay="0">
            <param name="posestamped" value="/vrpn_client_node/bebop10$(arg number)/pose"/> 
            <param name="tfname" value="bebop10$(arg number)/vrpn/base_link"/>
        </node>
        
        <node pkg="bebop_aruco" type="log_error.py" name="log_error" output="screen">
            <param name="input_vrpn_topic" value="/vrpn_client_node/bebop10$(arg number)/pose"/> 
            <param name="log_path" value="$(find bebop_aruco)/data/log"/> 
        </node>
        
        <node pkg="bebop_aruco" type="pc_camera_debug.py" name="debug" output="screen"/>
        <param name="camera_matrix_csv_path" value="$(find bebop_aruco)/data/camera/cameraMatrix.csv"/>
        <param name="dist_coeffs_csv_path" value="$(find bebop_aruco)/data/camera/distCoeffs.csv"/>
        
        <node name="image_republish" pkg="image_transport" type="republish" args="raw compressed">
            <remap from="in" to="image_raw" />
            <remap from="out" to="image_raw" />
        </node>
        
        <node pkg="rostopic" type="rostopic" name="camera_control" args="pub -1 camera_control geometry_msgs/Twist -- '[0.0, 0.0, 0.0]' '[0.0, -90, 0.0]'"/>
    </group>




</launch>